src/chains/ethereum/ethereum/tests/provider.test.ts

```diff
@@ -244,4 +245,40 @@ describe("provider", () => {
       );
     });
   });
+
+  describe("web3 compatibility", () => {
+    const networkId = 1234;
+    let provider: EthereumProvider;
+    let web3: Web3;
+    let accounts: string[];
+
+    beforeEach(async () => {
+      provider = await getProvider({
+        chain: { networkId },
+        wallet: { deterministic: true }
+      });
+      web3 = new Web3();
+      // TODO: remove "as any" once we've fixed our typing issues with web3
```

Comment:
Can you add a reference to the PRs that will fix this?

Reply:
```suggestion
      // TODO: remove "as any" once we've fixed our typing issues
      // with web3 (https://github.com/ChainSafe/web3.js/pull/4761)
```